v2/reader.go

```diff
@@ -0,0 +1,93 @@
+package car
+
+import (
+	"bufio"
+	"fmt"
+	blockstore "github.com/ipfs/go-ipfs-blockstore"
+	"io"
+
+	carv1 "github.com/ipld/go-car"
+	"github.com/ipld/go-car/v2/carbs"
+)
+
+const (
+	version2   = 2
+	indexCodec = carbs.IndexSorted
+)
+
+type (
+	Reader struct {
+		Header Header
+		r      io.ReaderAt
+	}
+)
+
+func NewReader(r io.ReaderAt) (*Reader, error) {
+	cr := &Reader{
+		r: r,
+	}
+	if err := cr.readPrefix(); err != nil {
+		return nil, err
+	}
+	if err := cr.readHeader(); err != nil {
+		return nil, err
+	}
+	return cr, nil
+}
+
+func (r *Reader) readPrefix() (err error) {
+	prefixSection := r.prefixReader()
+	header, err := carv1.ReadHeader(bufio.NewReader(prefixSection))
+	if err != nil {
+		return
+	}
+	if header.Version != version2 {
+		err = fmt.Errorf("invalid car version: %d", header.Version)
+	}
+	return
+}
+
+func (r *Reader) prefixReader() *io.SectionReader {
+	return io.NewSectionReader(r.r, 0, PrefixBytesSize)
+}
+
+func (r *Reader) readHeader() (err error) {
+	headerSection := r.headerReader()
+	_, err = r.Header.ReadFrom(headerSection)
+	return
+}
+
+func (r *Reader) headerReader() *io.SectionReader {
+	return io.NewSectionReader(r.r, PrefixBytesSize, HeaderBytesSize)
+}
+
+func (r *Reader) CarV1() (*carv1.CarReader, error) {
+	carV1Section := r.carV1Reader()
+	return carv1.NewCarReader(bufio.NewReader(carV1Section))
```

Comment:
why put an io.SectionReader through a bufio.Reader? if it's necessary, I'd document that briefly

Reply:
Typo; my bad. thanks